docs/modules/ROOT/pages/platform-deploy.adoc

```diff
@@ -0,0 +1,87 @@
+= Using with OpenZeppelin Platform
+
+The Hardhat Upgrades package can use the OpenZeppelin Platform for deployments instead of ethers.js.
+
+[[configuration]]
+== Configuration
+
+Create a deployment environment on the OpenZeppelin Platform and provide the Team API Key and secret in your `hardhat.config.js` or `hardhat.config.ts` file under `platform`:
+
+[source,js]
+----
+module.exports = {
+  platform: {
+    apiKey: process.env.API_KEY,
+    apiSecret: process.env.API_SECRET,
+  }
+}
+----
+
+[[usage]]
+== Usage
+
+When using the xref:api-hardhat-upgrades.adoc[Hardhat Upgrades API functions], enable OpenZeppelin Platform deployments using any of the ways below.
+
+NOTE: Only functions that have the `platform` option in their API reference support deployments through the OpenZeppelin Platform. If you enable the following but use functions that do not support `platform`, those functions will deploy using ethers.js.
```

Comment:
Scripts are meant to be reusable between a local dev network, testnets, and production.  The recommendation would be to set `deploy: true` in the config when using testnets/production (where platform will be used), so that the scripts don't need to be modified from when using a local network.

So there could be two modes:
- If using `platform.<function>` or using the option `{platform: true}`, then give an error if the function does not support platform.
- If using `deploy: true`, then just fall back if the function does not support platform, as described in the note.  If the function does support platform, but platform is not available (e.g. API key is not set), it would still give an error.

Reply:
That sounds reasonable.